ocs_ci/utility/utils.py

```diff
@@ -2536,6 +2540,15 @@ def ceph_health_check_base(namespace=None):
     namespace = namespace or config.ENV_DATA["cluster_namespace"]
     health = run_ceph_health_cmd(namespace)
 
+    if "daemons have recently crashed" in health and resolve_daemon_crash:
+        # importing here to avoid circular imports
+        from ocs_ci.ocs.resources.pod import get_ceph_tools_pod
+
+        log.error(health)
+        ct_pod = get_ceph_tools_pod()
+        ceph_crash_info_display(ct_pod)
```

Comment:
in fucntion of fix_ceph_health I would throw new type of exception like:
CephHealthForciblyRecoveredException

And in ceph_health_check this exception will not be handled by re-try as we need to raise this exception to actually fail at least one test case for it to notice the issue.

Also I think that we should at the end of this function check ceph health status with: fix_ceph_health=False

We can reduce - re-tries if needed for this call - just to give a chance that next test case will be fine to run with Ceph Health OK